test/includes/microcloud.sh

```diff
@@ -904,7 +937,8 @@ create_system() {
       exec > /dev/null
     fi
 
-    lxc init ubuntu-minimal:22.04 "${name}" --vm -c limits.cpu=2 -c limits.memory=4GiB
+    # ubuntu-minimal:24.04 does not execute cloud-init to enable the READY status. Still working on finding why this happens...
+    lxc init ubuntu:24.04 "${name}" --vm -c limits.cpu=2 -c limits.memory=4GiB
```

Comment:
Is this new? How are other PR tests working then?

Reply:
Not sure about the `-minimal` vs `non-minimal` part but 24.04 was picked as it doesn't use the `-kvm` kernel which is lacking the `dm-crypt` module. With 24.04's kernel, all the necessary modules are easily loadable.

Reply:
Ah I missed the change to `24.04`.

Reply:
I'm facing an issue with `ubuntu-minimal:24.04` in regards to `cloud-init`: strangely, with the minimal flavor, the `ready.sh` script is not executed:
```
cat << EOF | lxc profile edit default
config:
  cloud-init.user-data: |
    #cloud-config
    write_files:
      - content: |
          #!/bin/sh
          exec curl --unix-socket /dev/lxd/sock lxd/1.0 -X PATCH -d '{"state": "Ready"}'
        path: /var/lib/cloud/scripts/per-boot/ready.sh
        permissions: "0755"
EOF
```

whereas it works fine with `ubuntu:24.04`

Reply:
@gabrielmougard I cannot reproduce with `ubuntu-minimal-daily:24.04` here:
```
lxc init ubuntu-minimal-daily:24.04 foo --vm -c limits.cpu=2 -c limits.memory=4GiB
cat << EOF | lxc config set foo cloud-init.user-data -
#cloud-config
write_files:
  - content: |
     #!/bin/sh
     exec curl --unix-socket /dev/lxd/sock lxd/1.0 -X PATCH -d '{"state": "Ready"}'
    path: /var/lib/cloud/scripts/per-boot/ready.sh
    permissions: "0755"
EOF
lxc start foo
```

```
$ lxc ls foo
+------+-------+-----------------------+------------------------------------------------+-----------------+-----------+
| NAME | STATE |         IPV4          |                      IPV6                      |      TYPE       | SNAPSHOTS |
+------+-------+-----------------------+------------------------------------------------+-----------------+-----------+
| foo  | READY | 172.24.26.31 (enp5s0) | 2001:470:b1c3:7946:216:3eff:fe0d:bae8 (enp5s0) | VIRTUAL-MACHINE | 0         |
+------+-------+-----------------------+------------------------------------------------+-----------------+-----------+
```

Reply:
Ok then I'll try with `ubuntu-minimal-daily:24.04` and see

Reply:
`ubuntu-minimal:24.04` worked for me too but since it's a much older image, I think we should switch to `ubuntu-minimal-daily:24.04` if that also works for you.

Reply:
fixed

Reply:
Could you give it another try with `ubuntu-minimal-daily:24.04`, please? It was marked as fixed but maybe some rebase lost the change.

Reply:
@gabrielmougard how did it go with `ubuntu-minimal-daily:24.04`?

Reply:
fixed

Reply:
@simondeziel I can reproduce:
```bash
lxc init ubuntu-minimal-daily:24.04 foo --vm -c limits.cpu=2 -c limits.memory=4GiB
cat << EOF | lxc config set foo cloud-init.user-data -
#cloud-config
write_files:
  - content: |
     #!/bin/sh
     exec curl --unix-socket /dev/lxd/sock lxd/1.0 -X PATCH -d '{"state": "Ready"}'
    path: /var/lib/cloud/scripts/per-boot/ready.sh
    permissions: "0755"
EOF
lxc start foo
```

but the CI still seems to complain even with `ubuntu-minimal-daily:24.04` ...